packages/@aws-cdk/aws-ecs/lib/container-image.ts

```diff
@@ -1,17 +1,28 @@
 import ecr = require('@aws-cdk/aws-ecr');
+import secretsmanager = require('@aws-cdk/aws-secretsmanager');
 import cdk = require('@aws-cdk/cdk');
-
 import { ContainerDefinition } from './container-definition';
+import { CfnTaskDefinition } from './ecs.generated';
+
+/**
+ * Repository Credential resources
+ */
+export interface RepositoryCreds {
```

Comment:
I am wondering why we need this interface which adds another unneeded indirection instead of flat:

```ts
ContainerImage.fromRegistry('name', { credentialsSecret: secret });
```

Reply:
That is in fact how it works. This interface is not used for input, it's for output.